### DISCUSSION TITLE
Bundleless Mode?

### DISCUSSION THREAD

**Discussion by lsycxyj** [webpack/webpack#13442] [Ideas]
I think it will be much faster than "development" mode.

**Comment by alexander-akait** [comment]
Unfortunately it is not faster... memory cache is faster, but we start to work on ETag and other caches in webpack-dev-middleware, so maybe we can get closer to faster someday

**Reply by lsycxyj** [reply]
Doesn't Vite increase the bundling speed by building without bundle?

**Reply by alexander-akait** [reply]
> Doesn't Vite increase the bundling speed by building without bundle?

Not for all projects, it is always will be faster on simple repos, for some medium, for big projects memory/filesystem cache will be faster

Please do not forget deep analyze on dependencies (not only `import`/`require` in JS, also `@import`/`@use` in scss, `@import` in `CSS`, same for less and other types) requires parsing, yes it is possible to rewrite it on regexp, but this is not safe and will deprive many of the opportunities

I don't want to say it is impossible, it's a little more complicated and in most cases memory/filesystem cache is faster, gradually we will come to this

Vite is faster due `esbuild`, with webpack you can use https://github.com/privatenumber/esbuild-loader

**Reply by lsycxyj** [reply]
`esbuild` does majorly contribute the building speed, and works of deep analyses should be the same for bundled and bundleless. However, I suppose, for large projects, some bundles will still have to be re-bundled since some modules have been changed. Compared to bundless, it doesn't need to update things of those modules' parent bundle, which can save some efforts but also take advantages of caches.
Are there any benchmarks of "for some medium, for big projects memory/filesystem cache will be faster"?

**Reply by alexander-akait** [reply]
You can measure it yourself, also you can use `esbuild-loader` to improve cold initial build
